docs/docs/adrs/adr-012-separate-releasing.md

```diff
@@ -0,0 +1,67 @@
+---
+sidebar_position: 13
+title: Separate Releasing
+---
+# ADR 012: Separate Releasing
+
+## Changelog
+
+* {8/18/22}: Initial draft of idea in [#801](https://github.com/cosmos/interchain-security/issues/801)
+* {8/22/22}: Put idea in this ADR
+
+## Status
+
+Accepted
+
+## Context
+
+### Spike results
+
+I explored the idea of [#801](https://github.com/cosmos/interchain-security/issues/801) with this [spike branch](https://github.com/cosmos/interchain-security/tree/shawn%2Fgo-mod-split-aug-spike). Here's my conclusions:
+
+Splitting this repo to have multiple go.mods is possible. However there are various intricacies involved in decoupling the package hierarchy to have `x/ccv/types` as the lowest level dep, with `x/ccv/consumer` and `x/ccv/provider` being one dep layer above, with high-level tests depending on all three of the mentioned packages. I'd estimate this decoupling would take 2-5 workdays to finish, and require significant review effort.
+
+### Why go.mod split is not the way to go
+
+Let's take a step back and remember the issue we're trying to solve - **We need a clean way to decouple semver/releasing for the consumer and provider modules**. After more consideration, splitting up go.mods gives us little benefit in achieving this. Reasons:
+
+* The `go.mod` dependency system is tied to git tags for the entire repo (ex: `require github.com/cometbft/cometbft v0.37.2` refers to a historical tag for the entire cometbft repo).
+* It'd be an odd dev experience to allow modules to reference past releases of other modules in the same repo. When would we ever want the consumer module to reference a past release of the types module for example?
+* If we allow for `go.mod` replace statements to build from local source code, why split up the package deps at all?
+* Splitting go.mods adds a bunch of complexity with `go.work` files and all that shiz. VSCode does not play well with multiple module repos either.
+
+### Why separate repos is cool but also not the way to go
+
+All this considered, the cleanest solution to decoupling semver/releasing for the consumer and provider modules would be to have multiple repos, each with their own go.mod (3-4 repos total including high level tests). With this scheme we could separately tag each repo as changes are merged, they could share some code from `types` being an external dep, etc.
+
+I don't think any of us want to split up the monorepo, that's a lot of work and seems like bikeshedding. There's another solution that's very simple..  
+
+## Decision
+
+Slightly adapting [the current semver ruleset](https://github.com/cosmos/interchain-security/blob/cca008d856e3ffc60ec1a486871d0faa702abe26/CONTRIBUTING.md#semantic-versioning):
+
+* A library API breaking change to EITHER the provider or consumer module will result in an increase of the MAJOR version number for BOTH modules (X.y.z-provider AND X.y.z-consumer).
```

Comment:
I don't understand how this would work in practice, can you walk me through the release process?

This is how I understood this text:

Current version is `v5.0.0`

We make a statebreaking change on the consumer module that results in `v5.1.0-consumer`.

The released versions now become: `v5.0.0` and `v5.1.0-consumer`?

Alternatively, are we releasing as `v5.0.0-consumer` and `v5.0.0-provider` from the get go?

Can you add a quick visualization (using a basic markdown table) that shows what happens in which scenario so the proposed change can be understood more easily?

Reply:
One thing to be noted is that if we release a feature that affects both the provider and consumer, then it may end up in different versions, e.g., ICS v4.3.0-provider and v4.0.0-consumer. That would be a bit annoying, but I don't see any better way to deal with this. IMO, we should go ahead and merge this.

Reply:
if I understood well there will be no ICS versions anymore only x.y.z-provider and xy.z-consumer.
In that case it would be good to have a statement about compatibility between versions which can diverge from now on and maybe impact on testing.
From a user perspective, can I get from the versioning of provider and consumer which one 'should' work together?

Reply:
I think in the long term we should assume everything is backwards compatible, similar to ibc-go. In the short term there will be exceptions (like throttling v2), but we can communicate this clearly in the versioning descriptions.